### PR TITLE
DP-19089 Add noindex and follow to doc pages

### PR DESCRIPTION
**Pull request by danstafford**
**Description:**
Add noindex and follow to the robots metatag when "docs" is the second part of the url and "download" is not the last part. 

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-19089

**To Test:**
- [ ] Load a page like this one http://mass.local/doc/reopening-massachusetts
- [ ] Verify the robots metatag appears as described

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/modules/custom/mass_metatag/mass_metatag.module [hunk 1]
```diff
@@ -367,8 +367,17 @@ function mass_metatag_page_attachments(array &$attachments) {
   $current_uri = Url::fromRoute('<current>')->toString();
   $uri_parts = explode('/', $current_uri);
   $uri_end = end($uri_parts);
+  $noindex_and_follow = FALSE;
+
+  if ($uri_parts[1] === 'doc' && $uri_end !== 'download') {
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/mass_metatag.module]: This looks brittle to me. Can we get the route and see if its a media entity (and not a download route). This API-based condition will be more robust that looking and uri parts.

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module]: Thanks @weitzman , are you thinking something like this?
```
  if (\Drupal::routeMatch()->getRouteName() === 'entity.media.canonical' && $uri_end !== 'download') {
    $noindex_and_follow = TRUE;
  }
```

Comment (weitzman) [reply docroot/modules/custom/mass_metatag/mass_metatag.module]: Yes. And is `media_entity_download.download` the /download rooute? See docroot/modules/contrib/media_entity_download/media_entity_download.routing.yml

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module]: Hmmm... when a user hits the `/media/{media}/download` route they are served a file, not an html page where metatags can be added. So I don't think `mass_metatag_page_attachments` even runs in that case. Maybe we don't need the "download" check altogether?

Comment (weitzman) [reply docroot/modules/custom/mass_metatag/mass_metatag.module]: That makes sense to me ... Maybe do a git blame on that line and see if you can find out what issue introduced the /download check.

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module]: OK I removed the extraneous /download check. Now just media entities get the robots metatag and there are no fragile url part checks for this.

Comment (weitzman) [reply docroot/modules/custom/mass_metatag/mass_metatag.module]: Thanks. We should add a test IMO.

Comment (danstafford) [reply docroot/modules/custom/mass_metatag/mass_metatag.module]: Thanks @weitzman , I added a test for this when you get a minute.

**Pull request merged**